src/components/Todolist.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+import { Todo } from '../types/Todo';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  loadingTodoIds: number[];
+  setLoadingTodoIds: React.Dispatch<React.SetStateAction<number[]>>;
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  loading: boolean;
+  handleCompletedTodo: (todo: Todo) => void;
+  deleteTodo: (id: number) => void;
+  updateTodoTitle: (id: number, newTitle: string) => void;
+};
+
+export const List: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  loading,
+  loadingTodoIds,
+  handleCompletedTodo,
+  setLoadingTodoIds,
+  setLoading,
+  deleteTodo,
+  setErrorMessage,
+  updateTodoTitle,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <TodoItem
+          key={todo.id}
```

Comment:
better use uuid library

Reply:
look

Reply:
I completely agree, I use it, but in this project the tests crash when I add a library.